### PR TITLE
Add Cosmos DB PrivateEndpointConnection and PrivateLink Swagger

### PR DESCRIPTION
**Pull request by ayanMSFT**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [ ] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

File: specification/cosmos-db/resource-manager/Microsoft.DocumentDB/stable/2015-04-08/cosmos-db.json [hunk 1]
```diff
@@ -5896,6 +6073,109 @@
...
+    "PrivateLinkResource": {
+      "type": "object",
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/PrivateLinkResourceProperties"
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "#/definitions/Resource"
```
Comment (anthony-c-martin) [inline specification/cosmos-db/resource-manager/Microsoft.DocumentDB/stable/2015-04-08/cosmos-db.json]: Again, I think you'll want to extend ProxyResource here, and not Resource.

File: specification/cosmos-db/resource-manager/Microsoft.DocumentDB/stable/2015-04-08/examples/CosmosDBPrivateLinkResourceListGet.json [hunk 2]
```diff
@@ -0,0 +1,28 @@
+{
+  "parameters": {
+    "subscriptionId": "subid1",
+    "resourceGroupName": "rg1",
+    "accountName": "ddb1",
+    "api-version": "2015-04-08"
```
Comment (anthony-c-martin) [inline specification/cosmos-db/resource-manager/Microsoft.DocumentDB/stable/2015-04-08/examples/CosmosDBPrivateLinkResourceListGet.json]: So you're not releasing a preview version of this resource prior to GA? Just want to make sure you're aware of the implications of releasing it with a stable api version. See https://armwiki.azurewebsites.net/api_contracts/APIDeprecationPolicy.html for some more info on this.

File: specification/cosmos-db/resource-manager/Microsoft.DocumentDB/stable/2015-04-08/examples/CosmosDBPrivateEndpointConnectionGet.json [hunk 3]
```diff
@@ -0,0 +1,28 @@
+{
+  "parameters": {
+    "subscriptionId": "subid",
```
Comment (anthony-c-martin) [inline specification/cosmos-db/resource-manager/Microsoft.DocumentDB/stable/2015-04-08/examples/CosmosDBPrivateEndpointConnectionGet.json]: [nit] Examples should resemble real data as closely as possible - please use a dummy guid for "subscriptionId"

**Review by anthony-c-martin — CHANGES_REQUESTED** [review]

File: specification/cosmos-db/resource-manager/Microsoft.DocumentDB/stable/2015-04-08/cosmos-db.json [hunk 4]
```diff
@@ -3647,6 +3647,172 @@
...
+        "x-ms-long-running-operation": true,
+        "x-ms-examples": {
+          "Deletes a private endpoint connection with a given name.": {
+            "$ref": "./examples/CosmosDBPrivateEndpointConnectionDelete.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.DocumentDB/databaseAccounts/{accountName}/privateLinkResources": {
+      "get": {
```
[the remark this answers is not in the document]
Comment (ayanMSFT) [reply specification/cosmos-db/resource-manager/Microsoft.DocumentDB/stable/2015-04-08/cosmos-db.json]: we are implementing that API in Cosmos DB RP. Should I wait for it to be deployed before updating the swagger, and updating this PR?

[the remark this answers is not in the document]
Comment (KrisBash) [reply specification/cosmos-db/resource-manager/Microsoft.DocumentDB/stable/2015-04-08/cosmos-db.json]: @ayanMSFT  per review process, we want to see the swagger before exposing the API in the RP. The best way to do that is to flag the PR as doNotMerge, get it reviewed, and then ship the API, and finally merge the signed off PR

**Review by KrisBash — COMMENTED** [review]
There appear to be some blocking issues outstanding.

**Comment by ayanMSFT** [comment]
I will send another PR with these proposed API changes after following the review process

**Pull request closed without merging**